primitives/src/lib.rs

```diff
@@ -85,8 +85,17 @@ impl TryFrom<u8> for NodeType {
 	}
 }
 
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```

Comment:
Can you please write a documentation describing what the difference is between full, storage and cache storage modes?

I am wondering what a `Storage` storage mode is? might be redundant